invoices/invoiceregistry.go

```diff
@@ -74,6 +75,11 @@ type RegistryConfig struct {
 	// KeysendHoldTime indicates for how long we want to accept and hold
 	// spontaneous keysend payments.
 	KeysendHoldTime time.Duration
+
+	// SettlementInterceptor is a service that intercepts invoices during
+	// the settlement phase, enabling a subscribed client to determine the
+	// settlement outcome.
+	SettlementInterceptor *SettlementInterceptor
```

Comment:
The pattern we've been using elsewhere is to wrap this in an `fn.Option`, then you map over it, which gives logic that's a bit more linear.

Reply:
If I use `Option` here then my IDE will complain that we might be copying the `sync.WaitGroup` type field contained within `SettlementInterceptor`.

This is how I'm trying to use the `Option` type:
```golang
	SettlementInterceptor fn.Option[SettlementInterceptor]
```

Reply:
what about `fn.Option[*SettlementInterceptor]`

Reply:
I think we should put considerations like this aside until the PoC is complete.